File: src/services/PaymentService.js
import API from "../api/api";

export const makePayment = async (payment) => {
    const response = await API.post("/payments", payment);
    return response.data;
};

export const getPaymentById = async (id) => {
    const response = await API.get(`/payments/${id}`);
    return response.data;
};

export const getPaymentsByBooking = async (bookingId) => {
    const response = await API.get(`/payments/booking/${bookingId}`);
    return response.data;
};

File: payment-service/src/main/java/com/planmytrip/paymentservice/service/PaymentService.java
package com.planmytrip.paymentservice.service;


import com.planmytrip.paymentservice.entity.Payment;
import java.util.List;

public interface PaymentService {

    Payment makePayment(Payment payment);

    List<Payment> getPaymentsByBooking(Long bookingId);

    Payment getPaymentById(Long id);
}
